### PR TITLE
Case functions support: LOWER, UPPER

### PR DESCRIPTION
**Pull request by galkk**
Issue #128 

*Description of changes:*
* Added 2 new functions LOWER and UPPER that receive field name as a first parameter and, potentially, locale 

* Changed identifier generation strategy to id per function name instead of global id

While I feel that SQLFunctions should be refactored, to expedite the process as of now I'm adding just functions implemented by analogy with trim() and substring() functions

### Testing
integration testing, testing via kibana

Itegration tests include ORDER BY and GROUP BY clauses to ensure that everything is working.
Found 2 new issues that will be addressed separately: #176 , #175 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java [hunk 1]
```diff
@@ -109,6 +104,59 @@ public void functionAlias() throws Exception {
...
+    }
+
+    @Test
+    public void caseChangeWithAggregationTest() throws IOException {
+        String query = "SELECT UPPER(e.firstname), COUNT(*)" +
+                "FROM elasticsearch-sql_test_index_account/account e " +
+                "WHERE LOWER(e.lastname)='duke' " +
+                "GROUP BY UPPER(e.firstname) ";
+
+        System.out.println(executeQuery(query));
```
Comment (penghuo) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java]: remove this.

Comment (galkk) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java]: Done

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java [hunk 2]
```diff
@@ -74,6 +76,20 @@
             utilityFunctions)
             .flatMap(Set::stream).collect(Collectors.toSet());
 
+    private HashMap<String, Integer> generatedIds = new HashMap<>();
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Use interface `Map` instead?

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java [hunk 3]
```diff
@@ -74,6 +76,20 @@
...
+
+    /**
+     * Generates next id for given method name. The id's are increasing for each method name, so
+     * nextId("a"), nextId("a"), nextId("b") will return a_1, a_2, b_1
+     */
+    public String nextId(String methodName) {
```
Comment (penghuo) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: I think currently, the methodName is cached in request level. If i am right, does this cached method could be shared by different request?

Comment (galkk) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: No. The SqlFunctions is per-request, and this is instance field

Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Not sure if this look clear to you but it seems work for your purpose:

```
  private String nextId(String methodName) {
    return methodName + generatedIds.merge(methodName, 1, Integer::sum);
  }
```

Comment (galkk) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: Nice, i like it, thanks, will do

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java (hunk 2 above)
Comment (galkk) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java]: This is private field, so knowing detailed type could be beneficial in common case, but here Map will work as well, changed.

**Pull request merged**